substrate/bin/node/runtime/src/lib.rs

```diff
@@ -3303,18 +3303,49 @@ impl_runtime_apis! {
 		}
 
 		fn trace_block(
-			_block: Block,
-			_config: pallet_revive::evm::TracerConfig
+			block: Block,
+			config: pallet_revive::evm::TracerConfig
 		) -> Result<Vec<(u32, pallet_revive::evm::EthTraces)>, sp_runtime::DispatchError> {
-			unimplemented!()
+			let mut tracer = pallet_revive::debug::Tracer::from_config(config);
+			let mut traces = vec![];
+
+			pallet_revive::using_tracer(&mut tracer, || {
+				Executive::initialize_block(block.header());
+
+				for (index, ext) in block.extrinsics().into_iter().enumerate() {
+					let _ = Executive::apply_extrinsic(ext.clone());
+					pallet_revive::with_tracer(|tracer| {
+						if !tracer.has_traces() {
```

Comment:
I don't follow this logic, shouldn't it be the other way around i.e, just take the traces if `tracer.has_traces() == true`?
I expect the `tx_traces` below that you push to empty/default...

Worth explaining, maybe I'm just slow ^^

Reply:
yes sorry this is messed up just finished debugging this will patch it up